packages/medusa-plugin-brightpearl/src/services/brightpearl.js

```diff
@@ -653,18 +653,31 @@ class BrightpearlService extends BaseService {
       return
     }
 
-    const [reservationItems] =
-      await this.inventoryService_.listReservationItems({
+    const [reservations] = await this.inventoryService_.listReservationItems(
+      {
         id: ids,
-      })
+      },
+      {},
+      {
+        transactionManager: this.activeManager_,
+      }
+    )
+
+    const reservationItems = reservations.filter(
+      (reservation) => !reservation.metadata?.bpRow
+    )
 
     const client = await this.getClient()
 
     const { order, lineItems } = await this.getOrderFromReservation_(
       reservationItems
     )
 
-    if (!order?.metadata?.brightpearl_sales_order_id || !lineItems?.length) {
+    if (
+      !order?.metadata?.brightpearl_sales_order_id ||
+      !lineItems?.length ||
+      !reservationItems.length
+    ) {
       return this.attemptRetryEvent(
```

Comment:
todo: `without an order or a reservation`wdyt? since you can have an order but if there is no reservation we have the same case